### ISSUE TITLE
Expired auth sessions not deleted from the database

### ISSUE THREAD

**Issue by SamTrackbase** [masenf/reflex-local-auth#6]
Sessions are only deleted from the database in the `do_logout` method on `LocalAuthState`. If a user never logs out, and the value of `_auth_token` in local storage persists longer than the expiry time (default 7 days), I get a database integrity error as the `_login` method attempts to add a new session with a session ID which already exists in the table. The `do_logout()` call ([line 82 in `local_auth.py`](https://github.com/masenf/reflex-local-auth/blob/30dbc75581c5d701488b0ec1e95f7721ebf624ab/custom_components/reflex_local_auth/local_auth.py#L82)) is never called because the session has expired and so `self.is_authenticated` is False (at least in development - perhaps Reflex has cleared the cache of this variable)

Possible fix: just remove the `is_authenticated` check before `do_logout()`?

```  
def _login(
        self,
        user_id: int,
        expiration_delta: datetime.timedelta = DEFAULT_AUTH_SESSION_EXPIRATION_DELTA,
    ) -> None:
        """Create an LocalAuthSession for the given user_id.

        If the auth_token is already associated with an LocalAuthSession, it will be
        logged out first.

        Args:
            user_id: The user ID to associate with the LocalAuthSession.
            expiration_delta: The amount of time before the LocalAuthSession expires.
        """
        self.do_logout()
        if user_id < 0:
            return
        self.auth_token = self.auth_token or self.router.session.client_token
        with rx.session() as session:
            session.add(
                LocalAuthSession(  # type: ignore
                    user_id=user_id,
                    session_id=self.auth_token,
                    expiration=datetime.datetime.now(datetime.timezone.utc)
                    + expiration_delta,
                )
            )
            session.commit()
```

**Comment by masenf** [comment]
Thanks for reporting this issue. I'm actually in the middle of rewriting the auth session handling so that I can use it for multiple auth providers, so I should have a fix for this in the next few weeks. 

If you wanted to submit a workaround patch in the meantime which does as you suggest, I could release a new version.

**Comment by Hazed2004** [comment]
@SamTrackbase Can you make PR please until @masenf rewrite the auth session handling, thanks

**Comment by SamTrackbase** [comment]
@Hazed2004 thanks for the reminder - PR #7 submitted.

**Comment by masenf** [comment]
Fixed in 0.2.1

Thanks @SamTrackbase @Hazed2004
